src/sentry/issues/escalating.py

```diff
@@ -178,3 +189,53 @@ def _extract_project_and_group_ids(groups: Sequence[Group]) -> Dict[int, List[in
         group_ids_by_project[group.project_id].append(group.id)
 
     return group_ids_by_project
+
+
+def get_group_daily_count(project_id: int, group_id: int) -> int:
+    """Return the number of events a group has had today"""
+    date_now = datetime.now().date()
+    start_date = date_now
+    end_date = date_now + timedelta(days=1)
+    query = Query(
+        match=Entity(EntityKey.Events.value),
+        select=[
+            Function("count", []),
+        ],
+        where=[
+            Condition(Column("project_id"), Op.EQ, project_id),
+            Condition(Column("group_id"), Op.EQ, group_id),
+            Condition(Column("timestamp"), Op.GTE, start_date),
+            Condition(Column("timestamp"), Op.LT, end_date),
+        ],
+    )
+    request = Request(dataset=Dataset.Events.value, app_id=IS_ESCALATING_REFERRER, query=query)
+    return int(raw_snql_query(request, referrer=IS_ESCALATING_REFERRER)["data"][0]["count()"])
+
+
+def is_escalating(group: Group) -> bool:
+    """Return boolean depending on if the group is escalating or not"""
+    forecast_cache_key = f"escalating-forecast:{group.id}"
+    escalating_forecast = cache.get(forecast_cache_key)
+    date_now = datetime.now().date()
+    if escalating_forecast is None:
+        escalating_forecast = EscalatingGroupForecast.fetch(group.project.id, group.id)
+        escalating_forecast = (
+            escalating_forecast.date_added,
+            escalating_forecast.forecast,
+        )
+
+        # Set the cache to be valid until the next weekly escalating forecast task is run
+        forecast_cache_duration = (
+            (escalating_forecast[0] + timedelta(days=ONE_WEEK_DURATION)).date() - date_now
+        ).total_seconds()
+        cache.set(forecast_cache_key, escalating_forecast, forecast_cache_duration)
```

Comment:
I was thinking that we may want to do this as part of the `save` call in here:
https://github.com/getsentry/sentry/blob/0aaef53d7bf6924cf1410d7d4b72663b3116a7a6/src/sentry/issues/escalating_group_forecast.py#L39-L44